chia/full_node/full_node.py

```diff
@@ -1081,6 +1077,25 @@ async def validate_block_batches(
                 peer, blocks = res
                 start_height = blocks[0].height
                 end_height = blocks[-1].height
+
+                # in case we're validating a reorg fork (i.e. not extending the
+                # main chain), we need to record the coin set from that fork in
+                # fork_info. Otherwise validation is very expensive, especially
+                # for deep reorgs
+                peak: Optional[BlockRecord]
+                if fork_info is None:
+                    peak = self.blockchain.get_peak()
+                    extending_main_chain: bool = peak is None or (
+                        peak.header_hash == blocks[0].prev_header_hash or peak.header_hash == blocks[0].header_hash
+                    )
+                    if not extending_main_chain:
```

Comment:
maybe mention in a comment that it's important to *not* pass in a `ForkInfo` object when we're not validating a fork/in a reorg.